services/provider/server/server.go

```diff
@@ -4,46 +4,314 @@ import (
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to generate access token: %v", err)
+	}
+	return &pb.GenerateTokenResponse{AccessToken: token}, nil
 }
 
 // OnBoardConsumer RPC call to onboard a new OCS consumer cluster.
-func (c *ocsProviderServer) OnBoardConsumer(ctx context.Context, req *pb.OnBoardConsumerRequest) (*pb.OnBoardConsumerResponse, error) {
-	return &pb.OnBoardConsumerResponse{}, nil
+func (s *OCSProviderServer) OnBoardConsumer(ctx context.Context, req *pb.OnBoardConsumerRequest) (*pb.OnBoardConsumerResponse, error) {
+	// validate capacity
+	capacity, err := resource.ParseQuantity(req.Capacity)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "%q is not a valid resource capacity: %v", req.Capacity, err)
+	}
+
+	token, err := common.GetMetadata(ctx, "authorization")
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get on board consumer: %v", err)
+	}
+
+	uid, err := s.createStorageConsumer(req.ConsumerName, token, capacity)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to create storageConsumer resource %q: %v", req.ConsumerName, err)
+	}
+
+	consumer := consumer.NewConsumer(req.ConsumerName, uid, token)
+	err = s.authManager.consumerStore.Save(consumer)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to update consumer store: %v", err)
+	}
+
+	encryptedUID, err := s.authManager.GetEncryptedID(uid)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to encrypt consumer resource ID: %v", err)
+	}
+
+	// TODO: send correct granted capacity
+	return &pb.OnBoardConsumerResponse{StorageConsumerUUID: encryptedUID, GrantedCapacity: req.Capacity}, nil
 }
 
 // GetStorageConfig RPC call to onboard a new OCS consumer cluster.
-func (c *ocsProviderServer) GetStorageConfig(ctx context.Context, req *pb.StorageConfigRequest) (*pb.StorageConfigResponse, error) {
-	return &pb.StorageConfigResponse{}, nil
+func (s *OCSProviderServer) GetStorageConfig(ctx context.Context, req *pb.StorageConfigRequest) (*pb.StorageConfigResponse, error) {
+	consumerResource, err := s.getStorageConsumer(req.StorageConsumerUUID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to validate storage consumer: %v", err)
+	}
+
+	// Verify Status
+	switch consumerResource.Status.State {
+	case ocsv1alpha1.StorageConsumerStateFailed:
+		// TODO: get correct error message from the storageConsumer status
+		return nil, status.Errorf(codes.Internal, "storageConsumer status failed")
+	case ocsv1alpha1.StorageConsumerStateConfiguring:
+		return nil, status.Errorf(codes.Unavailable, "waiting for the rook resources to be provisioned")
+	case ocsv1alpha1.StorageConsumerStateDeleting:
+		return nil, status.Errorf(codes.Internal, "storageConsumer is already in deleting phase")
+	case ocsv1alpha1.StorageConsumerStateReady:
+		// use JSON connection string accordingly
+		_, err = s.getJSONConnectionDetails(consumerResource)
```

Comment:
result of this function call should be returned in StorageConfigResponse.

Reply:
updated